src/app/server/Server.h

```diff
@@ -465,14 +466,25 @@ class Server
 
         void OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex) override
         {
+            CommonPersistentData::FabricList fabricLinkedList;
+            FabricIndex next = kUndefinedFabricIndex;
+
             (void) fabricTable;
             ClearCASEResumptionStateOnFabricChange(fabricIndex);
             ClearSubscriptionResumptionStateOnFabricChange(fabricIndex);
 
             Credentials::GroupDataProvider * groupDataProvider = mServer->GetGroupDataProvider();
             if (groupDataProvider != nullptr)
             {
-                CHIP_ERROR err = groupDataProvider->RemoveFabric(fabricIndex);
+                CHIP_ERROR err = groupDataProvider->GetNextFabric(fabricIndex, next);
```

Comment:
I like the way you iterate the fabrics, but the `GetNextFabric()` method is not a good contribution to the interface. It only makes RemoveFabric more difficult to use. Now every time you want to remove a fabric, you need to run the GetNextFabric, RemoveFabric, UnregisterFabric combo, and their use is not evident.